File: DynamoDB-SDK-Examples/python/WorkingWithTables/export_table_s3.py
from __future__ import print_function  # Python 2/3 compatibility
import boto3
from datetime import datetime
from botocore.exceptions import ClientError
from dateutil.tz import tzlocal

# Export to S3 is only supported, if point in time recovery is enabled.
# Best practice is to externalize table name, can help with renaming of table if we want to restore later.
# Need to create S3 bucket and provide right set of permissions, good to enable version on S3.

s3_bucket_name = "YOUR S3 Bucket"
table_arn = "arn:aws:dynamodb:us-east-1:YOUR AWS ACCOUNT:table/Music"
table_name = "Music"

# Create low level client
session = boto3.session.Session()
client = session.client('dynamodb')

pitr_status = False
backup_response = client.describe_continuous_backups(TableName=table_name)
if backup_response.get("ContinuousBackupsDescription").get("PointInTimeRecoveryDescription").get("PointInTimeRecoveryStatus") == "ENABLED":
    pitr_status = True

print(f"PITR is enabled: {pitr_status}")

if pitr_status:
    try:
        response = client.export_table_to_point_in_time(
            TableArn=table_arn,
            ExportTime=datetime(2021, 12, 16, 16, 45, 0, tzinfo=tzlocal()),
            S3Bucket=s3_bucket_name,
            S3Prefix='DB_BACKUP',
            S3SseAlgorithm='AES256',
            ExportFormat='DYNAMODB_JSON'
        )
        print(response)
    except ClientError as e:
        print(e.response['Error']['Message'])

File: examples/SDK/node.js/WorkingWithTables/export_table_s3.js
// This is simple example to export dynamodb table to S3 bucket.

const {
  DynamoDBClient,
  ExportTableToPointInTimeCommand,
} = require("@aws-sdk/client-dynamodb");

// S3 bucket should be created before hand, and best practice is to enable encryption and version
// Make sure PITR is enabled in DYNAMODB table
// Make sure to replace with proper bucket name and account id
const s3_bucket_name = "YOUR S3 Bucket";
const table_arn = "arn:aws:dynamodb:us-east-1:YOUR ACCOUNT ID:table/Music";

async function exportTables() {
  const client = new DynamoDBClient({ region: "us-east-1" });
  try {
    return await client.send(
      new ExportTableToPointInTimeCommand({
        S3Bucket: s3_bucket_name,
        TableArn: table_arn,
        ExportFormat: "DYNAMODB_JSON",
        ExportTime: new Date("Thu Dec 16 2021 16:46:00 GMT-0600 (CST)"),
        S3Prefix: "DYNAMODB-Export",
        S3SseAlgorithm: "AES256",
      })
    );
  } catch (err) {
    console.error(err);
  }
}

exportTables()
  .then((data) =>
    console.log("ExportTables succeeded:", JSON.stringify(data, null, 2))
  )
  .catch((error) => console.error(JSON.stringify(error, null, 2)));
